### ISSUE TITLE
Potential false positive when exporting default function

### ISSUE THREAD

**Issue by rowellx68** [thepassle/eslint-plugin-barrel-files#5]
Plugin version: `1.0.3`
System: `macOS`

---

Setting the `barrel-files/avoid-barrel-files` rule with the following options seems to be treating files with default exported functions as barrel files. We've got a workaround without disabling the rule for the file or increasing the `amountOfExportsToConsiderModuleAsBarrel`.

Config:

```js
'barrel-files/avoid-barrel-files': [
  'error',
  { amountOfExportsToConsiderModuleAsBarrel: 0 }
],
```

Examples:

```tsx
// LazyLoadThisPage.tsx
export default function LazyLoadThisPage() {
  return (
    <></>
  );
}
```

```ts
// add.ts
export default function add(a: number, b: number): number {
  return a + b;
}
```

```ts
// some-config.ts
export default defineConfig({});
```

Workaround:

```tsx
// LazyLoadThisPage.tsx
function LazyLoadThisPage() {
  return (
    <></>
  );
}

export default LazyLoadThisPage;
```

```ts
// add.ts
function add(a: number, b: number): number {
  return a + b;
}

export default add;
```

```ts
// some-config.ts
const config = defineConfig({});

export default config;
```

**Comment by rowellx68** [comment]
Our intention is to completely get rid of/lint for barrel files so we set `amountOfExportsToConsiderModuleAsBarrel` to `0`. If we were to set it to `1` we would still be able to create them like so:

```ts
// LazyLoadThisPage/index.ts
export { LazyLoadThisPage } from './LazyLoadThisPage';
```

---

More than happy to do the changes and PR if you think this is a valid use case.

**Comment by thepassle** [comment]
sure, happy to take a PR
